File: frontend/types/respondent.ts
export interface Respondent {
  id: number;
  respondent_code: string;
  full_name: string;
  gender: string;
  age: number;
  occupation: string;
  created_at: string;
  updated_at: string;
}

export interface RespondentPagination {
  items: Respondent[];
  total: number;
  page: number;
  limit: number;
}

export interface CreateRespondentRequest {
  full_name: string;
  gender: string;
  age: number;
  occupation: string;
}

File: backend/app/schemas/respondent.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RespondentCreate(BaseModel):
    full_name: str
    gender: str
    age: int
    occupation: str


class RespondentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    respondent_code: str
    full_name: str
    gender: str
    age: int
    occupation: str
    created_at: datetime
    updated_at: datetime


# TAMBAHKAN INI
class RespondentPaginationResponse(BaseModel):
    items: list[RespondentResponse]
    total: int
    page: int
    limit: int